File: frontend/src/lib/api/analytics.ts
import apiClient from "./client";
import type { AnalyticsData, RiskItem } from "@/types/analytics";

export const analyticsApi = {
  getAnalytics: async (repoId: string): Promise<AnalyticsData> => {
    const response = await apiClient.get<AnalyticsData>(`/analytics/${repoId}`);
    return response.data;
  },

  getRisks: async (repoId: string): Promise<RiskItem[]> => {
    const response = await apiClient.get<RiskItem[]>(`/analytics/${repoId}/risks`);
    return response.data;
  },
};

File: backend/app/api/v1/analytics.py
"""Analytics endpoints."""
import uuid

from fastapi import APIRouter, HTTPException

from app.dependencies import CurrentUser, DBSession
from app.services.analytics_service import AnalyticsService
from app.schemas.analytics import AnalyticsResponse, RiskItem
from app.utils.exceptions import NotFoundError, UnauthorizedError

router = APIRouter()


@router.get("/{repo_id}", response_model=AnalyticsResponse)
async def get_analytics(repo_id: uuid.UUID, current_user: CurrentUser, db: DBSession):
    try:
        service = AnalyticsService(db)
        return await service.get_analytics(repo_id, current_user.id)
    except NotFoundError as e:
        raise HTTPException(status_code=404, detail=str(e))
    except UnauthorizedError as e:
        raise HTTPException(status_code=403, detail=str(e))


@router.get("/{repo_id}/risks", response_model=list[RiskItem])
async def get_risks(repo_id: uuid.UUID, current_user: CurrentUser, db: DBSession):
    try:
        service = AnalyticsService(db)
        return await service.get_risks(repo_id, current_user.id)
    except NotFoundError as e:
        raise HTTPException(status_code=404, detail=str(e))
    except UnauthorizedError as e:
        raise HTTPException(status_code=403, detail=str(e))
